src/js/common/components/SortableTable.jsx

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+
+class SortableTable extends React.Component {
+  constructor(props) {
+    super(props);
+    this.makeHeader = this.makeHeader.bind(this);
+    this.makeRow = this.makeRow.bind(this);
+  }
+
+  handleSort(value, order) {
+    return () => this.props.onSort(value, order);
+  }
+
+  makeHeader(field) {
+    // Determine what sort order the header will yield on the next click.
+    // By default, it will be ascending. Only make it descending if
+    // the current field is already sorted on in ascending order.
+    let nextSortOrder = 'ASC';
+    let sortIcon;
+
+    if (this.props.currentSort.value === field.value) {
+      if (this.props.currentSort.order === 'DESC') {
+        // Current icon will be descending.
+        // Next click will be ascending.
+        sortIcon = <i className="fa fa-caret-down"></i>;
+      } else {
+        // Current icon will be ascending.
+        // Next click will be descending.
+        sortIcon = <i className="fa fa-caret-up"></i>;
+        nextSortOrder = 'DESC';
+      }
+    }
+
+    return (
+      <th key={field.value}>
+        <a onClick={this.handleSort(field.value, nextSortOrder)}>
+          {field.label}&nbsp;&nbsp;&nbsp;{sortIcon}
```

Comment:
Using `&nbsp;&nbsp;&nbsp;`, which adds too much of an underline to the sort label. Add a `msg-sort` class to `sortIcon` and use CSS.